hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/MarkerBasedRollbackUtils.java

```diff
@@ -54,25 +64,35 @@ public static List<String> getAllMarkerPaths(HoodieTable table, HoodieEngineCont
                                                String instant, int parallelism) throws IOException {
     String markerDir = table.getMetaClient().getMarkerFolderPath(instant);
     FileSystem fileSystem = table.getMetaClient().getFs();
-    Option<MarkerType> markerTypeOption = MarkerUtils.readMarkerType(fileSystem, markerDir);
+    Option<MarkerType> markerTypeOption = readMarkerType(fileSystem, markerDir);
 
-    // If there is no marker type file "MARKERS.type", we assume "DIRECT" markers are used
+    // If there is no marker type file "MARKERS.type", first assume "DIRECT" markers are used.
+    // If not, then fallback to "TIMELINE_SERVER_BASED" markers.
```

Comment:
I inspected the code again to see if there will be a chance where marker type will be missing in timeline server based markers. [looks like ](https://github.com/apache/hudi/blob/ae7395df1e13112a1d99675edb04af05d0242703/hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/handlers/marker/MarkerDirState.java#L213) we first create the marker type file before  creating any markers. may be just the directory will exists w/o any marker type file and w/o any makers. But there should not be a case where marker type is missing and there are valid markers found incase of TimelineBasedMarkers.

Reply:
there is one flow where we could end up in this situation, but I could not piece end to end fully together. 
Lets say, rollback was triggered using timeline based markers. Towards the end, when everything is complete, we go ahead and delete the marker directory. During this, lets say marker type file was deleted, but before deleting all markers, process crashed. We might  end up in this state. But from looking at the code, rollback is actually complete by now (i.e. original commit files would have been deleted, and infact rollback commit would also be completed by now. So, a failure during deletion of marker directory should not have any impact. rollback will never be re-triggered for this case. 

Let me know if you guys can think of a scenario where this could happen. would really love to find the root cause.

Reply:
That's possible. However, it was reported in #5591 that the rollback failed.
The other thing is that the creation  of marker type file is contingent on existence of marker dir. What if the first time marker dir got created but marker type file was not created, then on retry `MarkerDirState#writeMarkerTypeToFile` sees that the marker dr already exists so it skips marker type file creation and sets `isMarkerTypeWritten` to true and goes ahead with writing marker files. For such cases, we need to explicitly check whether marker type file exists or not.

Reply:
that makes sense. thanks for finding the potential root cause.